packages/bun-usockets/src/crypto/openssl.c

```diff
@@ -100,9 +100,10 @@ struct us_internal_ssl_socket_context_t {
 struct us_internal_ssl_socket_t {
   struct us_socket_t s;
   SSL *ssl;
-  int ssl_write_wants_read; // we use this for now
-  int ssl_read_wants_write;
-  int pending_handshake;
+  unsigned int ssl_write_wants_read : 1; // we use this for now
+  unsigned int ssl_read_wants_write : 1;
+  unsigned int pending_handshake : 1;
+  unsigned int received_ssl_shutdown : 1;
```

Comment:
Do we use the non-opaque type in zig anywhere? I don’t think zig has a way to represent u1 in extern structs?